### PR TITLE
Revert FC workaround for ReflectionPad3d

### LINKED ISSUE #61248: functional API, volumetric reflection padding returns NotImplementedError
The following PR added `torch.nn.ReflectionPad3d` #59791.
The `torch.nn.functional` API should be updated as it can now use 3D reflection. It currently returns an `NotImplementError`.

According to #60837 this was disabled to work around FC issue

cc @albanD @mruberry @jbschlosser

### PR DESCRIPTION
**Pull request by jbschlosser**
Fixes #61248

### CODE REVIEW

**Review by ngimel — APPROVED** [review]

**Review by ngimel — COMMENTED** [review]
You should also probably remove type annotation from _nn.pyi.in https://github.com/pytorch/pytorch/pull/60837/files#diff-a93f4577904c757be5fd85988c2d9d6f56a77f92e65f30c0c00f84e9f8c4979aR35
